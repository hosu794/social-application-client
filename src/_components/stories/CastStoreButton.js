import React from "react";

import PropTypes from "prop-types";

function CastStoreButton({ id, castStore }) {
  return (
    <button onClick={castStore} className="button is-white">
      Like
    </button>
  );
}

CastStoreButton.propTypes = {
  storyActions: PropTypes.object,
  id: PropTypes.number,
};

export default CastStoreButton;
