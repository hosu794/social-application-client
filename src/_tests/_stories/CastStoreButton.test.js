import React from "react";

import configureStore from "redux-mock-store";

import { Provider } from "react-redux";

import { mount } from "enzyme";
import CastStoreButton from "../../_components/stories/CastStoreButton";
import { expectation } from "sinon";

let mockStore = configureStore([]);

let store = mockStore({});

const mockCallBack = jest.fn();

describe("Test for the CastStoreButton Component ", () => {
  let TestingComponent = () => (
    <Provider store={store}>
      <CastStoreButton castStore={mockCallBack} />
    </Provider>
  );

  let wrapper = mount(<TestingComponent />);

  test("should component render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should component be defined", () => {
    expect(CastStoreButton).toBeDefined();
  });

  test("should call function on button click", () => {
    wrapper.find("button").props().onClick();

    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
