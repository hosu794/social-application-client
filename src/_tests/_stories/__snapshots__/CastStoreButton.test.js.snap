// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the CastStoreButton Component  should component render correctly 1`] = `ReactWrapper {}`;
